internal/database/autocontact.go

```diff
@@ -0,0 +1,169 @@
+package database
+
+import (
+	"log/slog"
+	"net"
+	"os"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv4"
+)
+
+type SSHConfig struct {
+	User    string
+	Keypath string
+}
+
+func MonitorForDeadMachines(interval int, database Database, timespanForDeath int, l *slog.Logger, s SSHConfig) error {
+	downsampleTicker := time.NewTicker(time.Duration(interval))
+
+	for t := range downsampleTicker.C {
+		err := monitor(database, t, timespanForDeath, l, s)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func monitor(database Database, t time.Time, timespanForDeath int, l *slog.Logger, s SSHConfig) error {
+	last_seens, err := database.LastSeen()
+
+	if err != nil {
+		return err
+	}
+
+	for idx := range last_seens {
+		seen := last_seens[idx]
+
+		if seen.LastSeen < t.Add(-1*time.Duration(timespanForDeath)*time.Second).Unix() {
+			if ping(seen.Hostname, l) {
+				err := sshRestart(seen.Hostname, l, s)
+
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func sshRestart(remote string, l *slog.Logger, s SSHConfig) error {
+	user := s.User
+	keypath := s.Keypath
+
+	key, err := os.ReadFile(keypath)
+	if err != nil {
+		l.Error("Unable to read key: %v", err)
+		return err
+	}
+
+	signer, err := ssh.ParsePrivateKey(key)
+	if err != nil {
+		l.Error("Unable to parse key file %s: %v", keypath, err)
+		return err
+	}
+
+	config := &ssh.ClientConfig{
+		User: user,
+		Auth: []ssh.AuthMethod{ssh.PublicKeys(signer)},
+		// FIXME: Maybe be more secure??
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	client, err := ssh.Dial("tcp", remote+":22", config)
+	if err != nil {
+		l.Error("Failed to connect to %s: %v", remote, err)
+		return err
+	}
+	defer client.Close()
+
+	sess, err := client.NewSession()
+	if err != nil {
+		l.Error("Failed to create session: %v", err)
+		return err
+	}
+	defer sess.Close()
+
+	_, err = sess.Output("nohup ./data/gpuctl/satellite")
```

Comment:
This should be relative to `OnboardConf.DataDir`: 

https://github.com/gpuctl/gpuctl/blob/71ad89d9e52adfd0a5adb1a979e1de8e88021cbb/internal/webapi/server.go#L35

In addition, it should respect the convention of storing the logs, like onboard does.

https://github.com/gpuctl/gpuctl/blob/71ad89d9e52adfd0a5adb1a979e1de8e88021cbb/internal/onboard/onboard.go#L80-L81

It's probably a good idea to move this code from `database` to `onboard`, as that way we can share the ssh-start and ssh-restart functionality.